### PR TITLE
chore: update to new multiformats

### PR DESCRIPTION
**Pull request by achingbrain**
Replaces multihashes with the new multiformats module

BREAKING CHANGE: uses the CID class from the new multiformats module and `PeerID.createFromCID` now only takes `CID` instances

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -4,15 +4,21 @@
...
+const { base16 } = require('multiformats/bases/base16')
+const Digest = require('multiformats/hashes/digest')
 const cryptoKeys = require('libp2p-crypto/src/keys')
 const withIs = require('class-is')
 const { PeerIdProto } = require('./proto')
 const uint8ArrayEquals = require('uint8arrays/equals')
 const uint8ArrayFromString = require('uint8arrays/from-string')
 const uint8ArrayToString = require('uint8arrays/to-string')
 
+const IDENTITY_CODE = 0x00
```
[the remark this answers is not in the document]
Comment (achingbrain) [reply src/index.js:18]: I think one of the design goals behind the new `multiformats` module was to not have the big lookup tables that `multicodec` is the guardian of.

[the remark this answers is not in the document]
Comment (vasco-santos) [reply src/index.js:18]: Fair enough. Probably worth adding a reference to the table just to reference where they come from? Probably to https://github.com/multiformats/multicodec/blob/master/table.csv

[the remark this answers is not in the document]
Comment (achingbrain) [reply src/index.js:18]: In my journey through the codebase doing the multiformats stuff it seems that we only really use four or five of them.  I think the only weirdness is when bitswap has to [mess around with CID prefixes](https://github.com/ipfs/js-ipfs-bitswap/blob/58a8de1f4ffb8311bbbbe066e6a869df1d2bc4c0/src/types/message/index.js#L280) in incoming bitswap 1.1.0 messages.

[the remark this answers is not in the document]
Comment (vasco-santos) [reply src/index.js:18]: I think just adding a comment above with the multicodec table is enough here

[the remark this answers is not in the document]
Comment (achingbrain) [reply src/index.js:19]: Done!

**Review by vasco-santos — APPROVED** [review]
LGTM! Let me know when I should release it

**Comment by achingbrain** [comment]
I've tightened up the input types for `PeerId.createFromCID` as a follow on from https://github.com/multiformats/js-multiformats/pull/91#issuecomment-853762442 - it now only takes `CID` instances.

**Review by vasco-santos — APPROVED** [review]
👌🏼

**Comment by vasco-santos** [comment]
@achingbrain FYI #150 should also be ready soon (also breaking change), so I would like to get these shipped together as bubbling up PeerId updates across all libp2p stack is time consuming. This will create some conflicts that will need to be addresses by the second PR to be merged

File: src/index.d.ts [hunk 2]
```diff
@@ -94,6 +94,12 @@ declare namespace PeerId {
    * @param buf Protobuf bytes, as Uint8Array or hex-encoded string.
    */
   function createFromProtobuf(buf: Uint8Array | string): Promise<PeerId>;
+
+  /**
+   * Create PeerId from Protobuf bytes.
+   * @param buf Protobuf bytes, as Uint8Array or hex-encoded string.
```
Comment (vasco-santos) [inline src/index.d.ts:100]:
```suggestion
   * @param str encoded public key string.
```

**Review by vasco-santos — CHANGES_REQUESTED** [review]
Can you also have the README updated for `parse` and `createFromCID`?

**Review by vasco-santos — APPROVED** [review]

**Pull request merged**